Headlines/src/Managers/ImageCacheManager.swift

```diff
@@ -8,36 +8,40 @@
 
 import Foundation
 import UIKit
-import SDWebImage
+import Kingfisher
 
 class ImageCacheManager {
 
     static let shared = ImageCacheManager()
+    private let cache = ImageCache(name: "canillitapp.cache")
+    private var cachedImages = [String: UIImage]()
+    
 
-    var cachedImages = [String: UIImage]()
-
-    func image(forKey key: String) -> UIImage? {
+    func image(forKey key: String, completion: ((UIImage?) -> Void)? = nil) {
 
         // Look for it on cache
         if let cachedImage = cachedImages[key] {
-            return cachedImage
+            DispatchQueue.main.async {
+                completion?(cachedImage)
+            }
         }
 
         // Look for it on disk
-        if let cachedImage = SDWebImageManager.shared().imageCache?.imageFromDiskCache(forKey: key) {
-            // Cache it on memory
-            cachedImages[key] = cachedImage
-            return cachedImage
+        cache.retrieveImageInDiskCache(forKey: key) { [unowned self] result in
+            switch result {
+            case .success(let image):
+                self.cachedImages[key] = image
+                DispatchQueue.main.async {
+                    completion?(image)
+                }
+            case .failure(_):
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Canillitapp/headlines-iOS/blob/a5cc79d8b0bc8bc433e47d6a58e00624ade77840/Headlines/src/Managers/ImageCacheManager.swift#L37"></span>Arguments can be omitted when matching enums with associated types if they are not used.<br />
<code>empty_enum_arguments</code> <code>ImageCacheManager.swift:37</code></td>
    </tr>
  </tbody>
</table>